### PR TITLE
ntpsec: new port

### PR DESCRIPTION
**Pull request by lbschenkel**
From the homepage <https://www.ntpsec.org>:
> NTPsec project - a secure, hardened, and improved implementation of Network
> Time Protocol derived from NTP Classic, Dave Mills’s original.
>
> NTPsec, as its name implies, is a more secure NTP. Our goal is to deliver
> code that can be used with confidence in deployments with the most stringent
> security, availability, and assurance requirements.

This port is a drop-in replacement for ntp, with hardened internals. It should
behave identically in most circumstances, except that obsolete and/or insecure
features have been removed.

The current version of this port incorporates two newer patches from upstream
to fix a compilation error when leap second smearing is enabled.

The example file included in this port is the same example file included in
the ntp port, but with small adaptations for changes in ntpsec.

### CODE REVIEW

File: sysutils/ntpsec/Portfile [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+checksums           rmd160  9527cc6bc78f1a02c868fbb6a1fb583461b8fa81 \
+                    sha256  9d15ac59b67911eb1b42fe5916e8d00ded8b71ba34f1ab4fc0f251fc0d90d5e5
+
+depends_build       port:bison
+depends_lib         path:lib/libssl.dylib:openssl port:python27
+
+patchfiles          patch-macros.diff patch-smear.diff
+patch.pre_args      -p1
+
+set pythondir       "${frameworks_dir}/Python.framework/Versions/2.7/lib/python2.7/site-packages"
```
Comment (lbschenkel) [inline sysutils/ntpsec/Portfile]: Truth be told, I'm not entirely happy with this approach but this is what I have seen other ports doing.
Is there a better/idiomatic way of figuring out the PYTHONDIR for the default Python version without having to hardcode it like this?

Comment (Schamschula) [inline sysutils/ntpsec/Portfile]: You may be looking for the python port group:

PortGroup python 1.0

see under _resources/port1.0/group/python-1.0.tcl for details.

Comment (lbschenkel) [inline sysutils/ntpsec/Portfile]: I have changed the port to use the python port group, I hope it is slightly cleaner now.

**Comment by l2dy** [comment]
0.9.7 now and the patches seems to be integrated.

**Comment by tobypeterson** [comment]
@lbschenkel Should we take this now and await a new PR for 0.9.7, or will you update this one?

**Comment by lbschenkel** [comment]
Sorry, I am quite busy at the moment so I won't be able to update the PR right now (I don't mind if somebody else does). I'm neutral to merging/not merging now, so I'll leave it up to you.

**Comment by tobypeterson** [comment]
Let's do it. Update at your leisure.

**Pull request merged**